### PR TITLE
Account Summaries

### PR DESCRIPTION
**Pull request by dhakim87**
Administrators can now search by email to retrieve statistics about samples associated with a particular project

### CODE REVIEW

File: microsetta_admin/templates/email_stats_pulldown.html [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                        <th>{{col |e}}</th>
+                        {% endfor %}
+                    </tr>
+                </thead>
+                <tbody>
+                {% for row in resource.index %}
+                    <tr>
+                        {% for col in resource.columns %}
+                        {% if col == "account_id" %}
+                        <td><a href="{{ ui_endpoint }}/accounts/{{resource.at[row, col] |e}}">{{resource.at[row, col] |e}}</a></td>
```
Comment (wasade) [inline microsetta_admin/templates/email_stats_pulldown.html:73]: Indexing each value is going to be pretty inefficient. Would it make sense to use the dataframe's to_html method? see https://stackoverflow.com/a/45701949 or https://stackoverflow.com/a/58112545

File: microsetta_admin/upload_util.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    """
+    if file_name not in request.files or request.files['file'].filename == '':
+        return None, 'Must specify a valid file'
+
+    request_file = request.files[file_name]
+    try:
+        df = pd.read_csv(request_file, dtype=str)
+        col = df[col_name].tolist()
+    except Exception as e:  # noqa
+        return None, 'Could not parse barcodes file'
```
Comment (wasade) [inline microsetta_admin/upload_util.py:19]: I don't think this is assured to be a barcodes file :)

File: microsetta_admin/server.py [hunk 3]
```diff
@@ -227,6 +227,85 @@ def manage_projects():
...
+        )
+        if upload_err is not None:
+            return render_template('email_stats_pulldown.html',
+                                   **build_login_variables(),
+                                   resource=None,
+                                   search_error=[{'error': upload_err}])
+    else:
+        raise BadRequest()
+
+    if project is str:
```
Comment (wasade) [inline microsetta_admin/server.py:257]: I'm not sure I understand this. Do you mean `if isinstance(project, str)`? If so, I would recommend instead doing `if project is not None`

File: microsetta_admin/templates/email_stats_pulldown.html [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                {{error |e}}<br/>
+            {% endfor %}
+            </p>
+        {% endif %}
+        <script>
+           document.search_form.email.focus()
+        </script>
+    </form>
+    <br/>
+    Or upload a file and pull account summaries for all emails listed in the column "email"
```
Comment (wasade) [inline microsetta_admin/templates/email_stats_pulldown.html:46]: "Or upload a comma-separated value file (CSV) and pull..."

**Review by wasade — CHANGES_REQUESTED** [review]
As you noted, the table construction is probably pretty slow...

File: microsetta_admin/templates/email_stats_pulldown.html (hunk 1 above)
Comment (dhakim87) [reply microsetta_admin/templates/email_stats_pulldown.html:73]: That's pretty neat.  It's much simpler that way, though I'm not sure I can add a link to view the account this way.  I'll push up this version, let me know which you prefer.

Comment (wasade) [reply microsetta_admin/templates/email_stats_pulldown.html]: Here's one way we add in the account links (https://github.com/biocore/microsetta-admin/blob/master/microsetta_admin/templates/search_result.html#L52-L65), although it does differ some from the SO posts though with the benefit of avoiding a `.at`

**Pull request merged**